src/main/java/org/primefaces/component/ai/smartcam/SmartCamRenderer.java

```diff
@@ -0,0 +1,90 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.primefaces.component.ai.smartcam;
+
+import java.io.IOException;
+
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+import javax.faces.context.ResponseWriter;
+
+import org.primefaces.expression.SearchExpressionFacade;
+import org.primefaces.expression.SearchExpressionUtils;
+import org.primefaces.renderkit.CoreRenderer;
+import org.primefaces.util.WidgetBuilder;
+
+public class SmartCamRenderer extends CoreRenderer {
+
+    @Override
+    public void encodeEnd(FacesContext context, UIComponent component) throws IOException {
+        SmartCam cam = (SmartCam) component;
+
+        encodeMarkup(context, cam);
+        encodeScript(context, cam);
+    }
+
+    protected void encodeMarkup(FacesContext context, SmartCam cam) throws IOException {
+        ResponseWriter writer = context.getResponseWriter();
+        String clientId = cam.getClientId(context);
+        String style = cam.getStyle();
+        String styleClass = cam.getStyleClass();
+
+        writer.startElement("div", null);
+        writer.writeAttribute("id", clientId, null);
+        if (style != null) {
+            writer.writeAttribute("style", style, null);
+        }
+        if (styleClass != null) {
+            writer.writeAttribute("class", styleClass, null);
+        }
+
+        writer.endElement("div");
+    }
+
+    protected void encodeScript(FacesContext context, SmartCam component) throws IOException {
+        String clientId = component.getClientId(context);
+
+        WidgetBuilder wb = getWidgetBuilder(context);
+        wb.init("SmartCam", component.resolveWidgetVar(context), clientId)
+                .attr("width", component.getWidth(), 320)
+                .attr("height", component.getHeight(), 240)
+                .attr("renderTimeout", component.getRenderTimeout(), 50)
+                .attr("autoStart", component.isAutoStart(), true)
+                .callback("model", "function()", component.getModel())
+                .callback("predict", "function(model, input)", component.getPredict())
+                .callback("imageHandler", "function(video, canvasContext, detections)", component.getImageHandler());
+
```

Comment:
I think all `callbacks` should use `on` like onmodel, onpredict, and onimageHandler to follow PF conventions.

Why not make them `'p:ajax` events as well?